File: travel-catalog-api/src/main/java/com/certimetergroup/smart/travel/model/AgencyPackage.java
package com.certimetergroup.smart.travel.model;

import io.quarkus.mongodb.panache.common.MongoEntity;
import io.quarkus.mongodb.panache.reactive.ReactivePanacheMongoEntity;
import java.time.Instant;
import java.util.Set;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.NoArgsConstructor;
import shared.DestinationSummary;
import shared.PackageStatus;
import shared.Price;
import shared.UserSummary;
import shared.order.AccommodationOrder;
import shared.order.ActivityOrder;
import shared.order.FlightOrder;

@Builder
@NoArgsConstructor
@AllArgsConstructor
@MongoEntity(collection = "agency_packages")
public class AgencyPackage extends ReactivePanacheMongoEntity {

  public String name;
  public String description;
  public Set<String> tags;
  public PackageStatus status;
  public Instant startDate;
  public Instant endDate;
  public Price totalPrice;
  public DestinationSummary destination;
  public String mainPicture;
  public Set<String> pictures;
  public FlightOrder departureFlight;
  public FlightOrder returnFlight;
  public AccommodationOrder accommodation;
  public Set<ActivityOrder> activities;
  public UserSummary agentInfo;
  public Instant creationDate;
  public Integer quantity;    // Package target number of people
}

File: frontend/src/app/interfaces/model/AgencyPackage.ts
import { PackageStatus } from '../enums/PackageStatus';
import { AccommodationOrder } from '../orders/AccommodationOrder';
import { ActivityOrder } from '../orders/ActivityOrder';
import { FlightOrder } from '../orders/FlightOrder';
import { DestinationSummary } from './shared/DestinationSummary';
import { Price } from './shared/Price';
import { UserSummary } from './shared/UserSummary';

export interface AgencyPackage {
  id: string;
  name: string;
  description: string;
  tags: string[];
  status: PackageStatus;
  startDate: string;
  endDate: string;
  totalPrice: Price;
  destination: DestinationSummary;
  mainPicture: string;
  pictures?: string[];
  departureFlight: FlightOrder;
  returnFlight: FlightOrder;
  accommodation: AccommodationOrder;
  activities: ActivityOrder[];
  agentInfo: UserSummary;
  creationDate: string;
  quantity: number;
}

export interface PartialAgencyPackage {
  name: string;
  description: string;
  tags: string[];
  totalPrice: Price;
  mainPicture: string;
}
